src/libcollections/vec.rs

```diff
@@ -166,7 +166,7 @@ impl<T> Vec<T> {
         // non-null value which is fine since we never call deallocate on the ptr
         // if cap is 0. The reason for this is because the pointer of a slice
         // being NULL would break the null pointer optimization for enums.
-        Vec { ptr: unsafe { NonZero::new(EMPTY as *mut T) }, len: 0, cap: 0 }
+        Vec { ptr: unsafe { NonZero::new(EMPTY as *mut T).unwrap() }, len: 0, cap: 0 }
```

Comment:
Would be nice to have a function returning such a pointer - `EMPTY` has to be non-null anyways, so it should be tied in with `NonZero`.